### DISCUSSION TITLE
How to update firmware

### DISCUSSION THREAD

**Discussion by JohnFante22** [UzixLS/zx-sizif-512#23] [General]
I have a revision D board. And I am very happy with it :-)

Is there a guide to how I (safely) update the board to the latest firmware?

Thank you in advance!

**Comment by UzixLS** [comment]
To update CPLD firmware you need USB Blaster (about ~5$ on ali) and Quartus Programmer software (full Quartus distribution isn't required).
Step-by-step manual:

0. Connect USB blaster to PC, install drivers if needed;
1. Connect USB blaster to JTAG header on Sizif board;
2. Power on Sizif;
3. Run Quartus Programmer;
4. Press "Auto Detect" button, EPM1270 should appears in window;
5. Double click on "< none >" entry in window, choose firmware file (cpld.rev.X.pof)
6. Check "Program" and "Verify" checkboxes;
7. Press "Start" button;
8. Wait for "100% Successfull";
9. Done.
![изображение](https://user-images.githubusercontent.com/1556813/186500267-4b944c99-b8be-4664-997a-4db533a2fb78.png)

To update ROM you need some parallel EEPROM/Flash programmer, e.g. TL866 or any other.
Step-by-step manual:

0. Power off Sizif;
1. Take out ROM IC from Sizif;
2. Program firmware file (rom.rev.X.pof) guided by your programmer manual;
3. Take back IC to Sizif;
4. Done :)

**Reply by JohnFante22** [reply]
Thank you very much! Have the Blaster need the TL866. But that should be a minor task ... off we go :-) And thank you for a great device!

**Reply by JohnFante22** [reply]
I finally got around to try to updating. Got hold of a TL866 and a new blaster. The last one gave me issues. However, so far without succes :-(
Tried to update the CPLD but I simply can not get Quartus Programmer to see the CPLD. See screenshot below.
The blaster is working correctly - at least as far as I know - and I installed the latest version of Quartus programmer (23.1std1 Build 993 05/14/2024) and the latest availeble Blaster drivers (2.12.28.0 from 16-08-2017). I am running Windows 11. Also tried it on PopOS linux. Same result.
I have tried the JTAG Chain debugger as suggested but it only says:
"Error: JTAG chain problem detected
Error: No device detected".
I know this is not a  Quartus support forum but do you have any suggestions/advice?
![JTAG error](https://github.com/UzixLS/zx-sizif-512/assets/107305437/eeae61ef-6409-46da-b876-6bda54d5d3a6)

**Reply by UzixLS** [reply]
In 99% of cases, problem occurs due to a bad connection. The remaining 1% are software problems and so on.

**Reply by JohnFante22** [reply]
You where right, Apparantly I had placed the JTAG-cable the wrong way around.

The way that @adam-ainsworth had success with (and thank you for that 👍 ) was not the right way for me. I had to attacht it the other way round. 

Now on the updating the ROM ... fingers crossed. 

And finally a feature requst - now that I am here :-) ... I have always had a thing for the Opus Discovery disk interface. Would it be possible to make the Sizif emulate that so it can read OPD diskimages? I know it is far out but I just wanted to throw it out there. :-D ...

**Reply by UzixLS** [reply]
> And finally a feature requst - now that I am here :-) ... I have always had a thing for the Opus Discovery disk interface. Would it be possible to make the Sizif emulate that so it can read OPD diskimages? I know it is far out but I just wanted to throw it out there. :-D ...

It's out of my interest. Is there any unique software for Opus?

**Reply by JohnFante22** [reply]
Completely fair :-) 
Just think it is cool piece of kit. Sadly, VERY difficult to get now a days ...
Thank you for the Sizif 👍

**Comment by oO-cozy-Oo** [comment]
Can you just update the FW only or do you need to do the ROM as well , as the TL866 is about £76, I`ve ordered the Blaster as it was only £7

Cheers Cozy

**Reply by UzixLS** [reply]
You need to update both for correct work.

**Comment by adam-ainsworth** [comment]
Thank you very much for the walk through - I was just successful in installing the latest firmware and ROM. If anyone would like the links, I used these

https://www.terasic.com.tw/wiki/File:Usb_blaster_q16.1.zip (you may not need this as drivers come with the programmer software)
https://www.intel.com/content/www/us/en/software-kit/757262/intel-quartus-prime-lite-edition-design-software-version-22-1-for-windows.html?

Some help on finding the correct programmer download 

http://www.terasic.com.tw/wiki/Chapter_1_Download_and_install_Quartus_Programmer

Also, this is how you connect the JTAG connector cable (I wasn't sure which way up it went)

https://i2.wp.com/terenceang.com/wp-content/uploads/2021/08/20210819_160210687_iOS-scaled-1.jpg (thanks Terance!)

**Comment by GiacVan** [comment]
Hello there.
I have a rev D1  board and it doesn't work well.
In the menu,  when I try to change type of machine from the default Pentagon, all becomes frozen; this doesn't happen with all the other parameters, which I manage to change regularly : so it's impossible to change type of machine.
I updated firmware and ROM, but the problem wasn't solved.
Can somebody help me ?
